### PR TITLE
Daily challenge endpoints

### PR DESCRIPTION
**Pull request by santoshernandezr**
This PR is for the daily challenge functionality. The major changes include: 

1. When the user goes into the daily challenge page, check if they have completed the daily challenge. If so, render the game, else show them the completed daily challenge page.
2. Use cron job in the server to call two endpoints at midnight everyday. One endpoint to update the daily challenge words, and the second endpoint to reset all users daily challenge boolean (if they completed it) to false, so they can play the next day.
3. Endpoints to help with daily challenge tasks.
4. Removing node modules from the source code.

### CODE REVIEW

File: client/src/components/common/profiilePictureGrid.jsx
Comment (grrdsoto) [inline client/src/components/common/profiilePictureGrid.jsx:1]: profi*lePictureGrid.jsx

**Review by grrdsoto — CHANGES_REQUESTED** [review]
need to add node_modules to gitignore

**Comment by santoshernandezr** [comment]
> need to add node_modules to gitignore

added

File: client/src/components/common/profiilePictureGrid.jsx
Comment (santoshernandezr) [reply client/src/components/common/profiilePictureGrid.jsx:1]: What does this mean?

**Pull request merged**